pennylane/devices/default_tensor.py

```diff
@@ -686,10 +725,11 @@ def expval(self, measurementprocess: MeasurementProcess) -> float:
         """
 
         obs = measurementprocess.obs
+        return expval_core(obs, self)
 
-        result = self._local_expectation(qml.matrix(obs), tuple(obs.wires))
-
-        return result
+    def state(self, measurementprocess: MeasurementProcess):  # pylint: disable=unused-argument
+        """Returns the MPS state in vector form."""
+        return self._quimb_circuit.psi.to_dense().ravel()
```

Comment:
This seems to work even with `method="tn"`, so it would only be necessary to update the description